crates/red_knot_python_semantic/resources/mdtest/variables.md

```diff
@@ -0,0 +1,134 @@
+# Variables
+
+## Union resolution
+
+```py
+if flag:
+    x = 1
+else:
+    x = 2
+reveal_type(x)  # revealed: Literal[1, 2]
+```
+
+## Assignment
+
+### Walrus operator
+
+```py
+x = (y := 1) + 1
+reveal_type(x)  # revealed: Literal[2]
+reveal_type(y)  # revealed: Literal[1]
+```
+
+### Walrus self-addition
+
+```py
+x = 0
+(x := x + 1)
+reveal_type(x)  # revealed: Literal[1]
+```
+
+### Multi-target assignment
+
+```py
+x = y = 1
+reveal_type(x)  # revealed: Literal[1]
+reveal_type(y)  # revealed: Literal[1]
+```
+
+### Maybe unbound
+
+```py
+if flag:
+    y = 3
+x = y
+reveal_type(x)  # revealed: Unbound | Literal[3]
+```
+
+### Unbound
+
+```py
+x = foo; foo = 1
+reveal_type(x)  # revealed: Unbound
+```
+
+### Annotation only transparent to local inference
```

Comment:
All the tests from here onward should go in `shadowing.md` along with some other tests I commented above.